src/main/java/net/minecraftforge/registries/DeferredRegister.java

```diff
@@ -234,28 +184,17 @@ public TagKey<T> createTagKey(@NotNull String path)
      * @see #createTagKey(String)
      * @see #createOptionalTagKey(ResourceLocation, Set)
      */
-    @NotNull
     public TagKey<T> createTagKey(@NotNull ResourceLocation location)
     {
-        if (this.registryKey == null)
-            throw new IllegalStateException("The registry name was not set, cannot create a tag key");
         Objects.requireNonNull(location);
         return TagKey.create(this.registryKey, location);
     }
 
     /**
-     * Creates a tag key with the current modid and provided path that will use the set of defaults if the tag is not loaded from any datapacks.
-     * Useful on the client side when a server may not provide a specific tag.
-     * To control the namespace, use {@link #createOptionalTagKey(ResourceLocation, Set)}.
-     *
-     * @throws IllegalStateException If the registry name was not set.
-     * Use the factories that take {@link #create(ResourceLocation, String) a registry name} or {@link #create(IForgeRegistry, String) forge registry}.
-     * @see #createTagKey(String)
-     * @see #createTagKey(ResourceLocation)
+     * Calls {@link #createOptionalTagKey(ResourceLocation, Set)} with the given path, using the current modid as the namespace.
```

Comment:
As already described, I prefer duplication of information vs. forcing people to go through javadoc chains to find the information relevant to the method they are calling.